compiler/deSugar/DsArrows.hs

```diff
@@ -604,7 +604,8 @@ dsCmd ids local_vars stack_ty res_ty
     core_body <- dsExpr (HsCase noExt exp
                          (MG { mg_alts = L l matches'
                              , mg_arg_tys = arg_tys
-                             , mg_res_ty = sum_ty, mg_origin = origin }))
+                             , mg_res_ty = sum_ty, mg_origin = origin
+                             , mg_weight = Omega })) -- TODO: MattP
```

Comment:
Even though a moment inspection will make it clear that the problem is a rogue `Omega`, my advice is to make a note of this. This will help when you come back here in a week or two.